packages/base/src/viewlist.ts

```diff
@@ -68,9 +68,8 @@ export class ViewList<T> {
       this.views.length - first_removed
     );
     for (let j = 0; j < removed.length; j++) {
-      removed[j].then(function (view) {
-        remove.call(context, view);
-      });
+      const view = await removed[j];
```

Comment:
Doesn't this change the behaviour of the code, previously this function would return even before the promise `remoted[j]` resolves.
However now we're changing the behaviour, and forcing these promises to resolve before we can return from this function.

This could have some undesired consequenses

Reply:
I can definitely see how this could cause things to go wonky - particularly seems like it would slow down every update cycle significantly. Maybe this is one of the more obvious methods to keep as a promise.